src/routes/volume/detail/VolumeInfo.js

```diff
@@ -133,22 +133,49 @@ function VolumeInfo({ selectedVolume, snapshotModalState, engineImages, hosts, c
   }
 
   const volumeSizeEle = () => {
-    let oldSize = selectedVolume && selectedVolume.controllers && selectedVolume.controllers[0] && selectedVolume.controllers[0].size ? selectedVolume.controllers[0].size : ''
-    let isExpanding = (selectedVolume.size !== oldSize && selectedVolume.state === 'attached')
-    let message = `The volume is in expansion progress from size ${formatMib(oldSize)} to size ${formatMib(selectedVolume.size)}`
+    let expectedSize = selectedVolume.size
+    let currentSize = selectedVolume?.controllers[0]?.size ? selectedVolume.controllers[0].size : ''
+    let isExpanding = selectedVolume?.controllers[0] && parseInt(expectedSize, 10) !== parseInt(currentSize, 10) && selectedVolume.state === 'attached' && parseInt(currentSize, 10) !== 0
+    // The expected size of engine should not be smaller than the current size.
+    let expandingFailed = isExpanding && selectedVolume?.controllers[0]?.lastExpansionError !== ''
+    let message = ''
+    if (expandingFailed) {
+      message = (<div>
+        <div>Expansion Error: {selectedVolume?.controllers[0]?.lastExpansionError}</div>
```

Comment:
`selectedVolume?.controllers[0]?.lastExpansionError` already checked before, so basically you just use `selectedVolume..controllers[0].lastExpansionError`